### PR TITLE
community: Add you.com utility, update you retriever integration docs

### PR DESCRIPTION
**Pull request by scottnath**
- **Description: changes to you.com files** 
    - general cleanup
    - adds community/utilities/you.py, moving bulk of code from retriever -> utility
    - removes `snippet` as endpoint
    - adds `news` as endpoint
    - adds more tests

<s>**Description: update community MAKE file** 
    - adds `integration_tests`
    - adds `coverage`</s>

- **Issue:** the issue # it fixes if applicable,
    - [For New Contributors: Update Integration Documentation](https://github.com/langchain-ai/langchain/issues/15664#issuecomment-1920099868)
- **Dependencies:** n/a
- **Twitter handle:** @scottnath
- **Mastodon handle:** scottnath@mastodon.social

### CODE REVIEW

File: libs/community/Makefile [hunk 1]
```diff
@@ -6,6 +6,14 @@ all: help
 # Define a variable for the test file path.
 TEST_FILE ?= tests/unit_tests/
 
+# Run unit tests and generate a coverage report.
```
Comment (hwchase17) [inline libs/community/Makefile:9]: why is makefile changing? at very least should probably be separate pr

**Review by hwchase17 — COMMENTED** [review]
ideally we dont change the notebook name (eg lets keep it as you-retriever) so it doesnt break any links to it

**Review by hwchase17 — APPROVED** [review]
this is great - thanks!

**Comment by ktynski** [comment]
Is this broken for anyone else as of today? Im getting:

  File "/home/kristin/.cache/pypoetry/virtualenvs/trip-planner-_gQIwBJ6-py3.9/lib/python3.9/site-packages/langchain_community/retrievers/you.py", line 49, in _get_relevant_documents
    n_hits = self.n_hits or len(results["hits"])
KeyError: 'hits'

**Comment by scottnath** [comment]
hi @ktynski that line is from the older version of `retrievers/you.py`, this PR changed that file, which moved that section from the retriever to the utility. Please make sure you have the latest version of langchain to get the new changes.

cheers
